src/lib/mirage_block_ocaml.ml

```diff
@@ -10,7 +10,7 @@ module Log = (val Logs.src_log src : Logs.LOG)
 (* TODO: parameterise this over block implementations *)
 module Time = struct
   type 'a io = 'a Lwt.t
-  let sleep = Lwt_unix.sleep
+  let sleep_ns ns = Lwt_unix.sleep (Int64.to_float ns /. 1e9)
```

Comment:
You will probably be interested by https://hannesm.github.io/duration/doc/Duration.html#VALof_sec